src/test/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/integration/IncentiveLevelResourceTestBase.kt

```diff
@@ -66,22 +72,17 @@ class IncentiveLevelResourceTestBase : SqsIntegrationTestBase() {
     ).collect()
   }
 
-  private fun AmazonSQS.getApproxQueueSize(queueUrl: String): Int? =
-    getQueueAttributes(queueUrl, listOf("ApproximateNumberOfMessages"))
-      .attributes["ApproximateNumberOfMessages"]?.toInt()
-
   protected fun assertNoDomainEventSent() {
     val sqsClient = incentivesQueue.sqsClient
-    val queueSize = sqsClient.getApproxQueueSize(testDomainEventQueueUrl!!)
+    val queueSize = sqsClient.countMessagesOnQueue(testDomainEventQueueUrl!!).get()
     assertThat(queueSize).isEqualTo(0)
   }
 
   protected fun assertDomainEventSent(eventType: String): HMPPSDomainEvent {
     val sqsClient = incentivesQueue.sqsClient
-    val queueSize = sqsClient.getApproxQueueSize(testDomainEventQueueUrl!!)
-    assertThat(queueSize).isEqualTo(1)
+    await untilCallTo { sqsClient.countMessagesOnQueue(testDomainEventQueueUrl!!).get() } matches { it == 1 }
```

Comment:
Added this, and other `awaits`, because was finding tests were consistently failing

Assumed it was because we're now using `software.amazon.awssdk.services.sqs.SqsAsyncClient` (via `uk.gov.justice.service.hmpps:hmpps-sqs-spring-boot-starter:2.0.0-beta-14`) whereas previously we were using the synchronous class `com.amazonaws.services.sqs.AmazonSQS`